File: ts/src/leetcode/1920_build_array_from_permutation.ts
function buildArray(nums: number[]): number[] {
  const n = nums.length;

  for (let i = 0; i < n; i++) {
    const a = nums[i];
    const b = nums[a] % n;

    nums[i] = b * n + a;
  }

  for (let i = 0; i < n; i++) {
    nums[i] = Math.trunc(nums[i] / n);
  }

  return nums;
}

export default buildArray;

File: swift/1920_build_array_from_permutation.swift
class Solution {
    func buildArray(_ nums: [Int]) -> [Int] {
       var nums = nums
        let n = nums.count

        for i in 0 ..< n {
            let a = nums[i]
            let b = nums[a] % n

            nums[i] += (b * n)
        }

        for i in 0 ..< n {
            nums[i] /= n
        }

        return nums
    }
}

let solution = Solution()
var nums = [5, 3, 0, 1, 2, 4]
print(solution.buildArray(nums))
